doc/contributor/howto-guide-setup-cmake-environment.md

```diff
@@ -0,0 +1,125 @@
+# How-to Guide: Setup for CMake-based builds
+
+This document describes how to setup your workstation to build the Google Cloud
+C++ client libraries using CMake. The intended audience is developers of the
+client libraries that want to verify their changes will work with CMake and/or
+prefer to use CMake for whatever reason.
+
+* Packaging maintainers or developers that prefer to install the library in a
+  fixed directory (such as `/usr/local` or `/opt`) should consult the
+  [packaging guide](/doc/packaging.md).
+* Developers wanting to use the libraries as part of a larger CMake or Bazel
+  project should consult the [quickstart guides](/README.md#quickstart) for the library
+  or libraries they want to use.
+* Developers wanting to compile the library just to run some of the examples or
+  tests should read the [build and install](/README.md#building-and-installing)
+  section from the top-level README.
+* Contributors and developers to `google-cloud-cpp` wanting to work with CMake
+  should read the current document.
+
+The document assumes you are using a Linux workstation running Ubuntu, changing
+the instructions for other distributions or operating systems is left as an
+exercise to the reader. PRs to improve this document are most welcome!
+
+Unless explicitly stated, this document assumes you running these commands at
+the top-level directory of the project, as in:
+
+```shell
+cd $HOME
+git clone git@github:<github-username>/google-cloud-cpp.git
+cd google-cloud-cpp
+```
+
+## Install the dependencies in `$HOME/local-cpp`.
```

Comment:
This section covers more than just installing the deps, so a more general title may be good. Maybe something like "Installing with dependencies in `$HOME/local-cpp`"

Reply:
Broke the sections into smaller pieces.